File: bubble_sort.cpp
#include <iostream>

using namespace std;


int main(void) {
    int arr[10] = {9, 0, 1, 8, 2, 7, 6, 3, 5, 4};
    for(int i=0; i<10; i++) {
        for(int j=0; j<10-i-1; j++) {
            if(arr[j] > arr[j+1]) {
                int temp = arr[j+1];
                arr[j+1] = arr[j];
                arr[j] = temp;
            }
        }
    }

    for(int k=0; k<10; k++) {
        cout << *(arr + k) << " ";
    }
    cout << endl;
    return 0;
}

File: golang/bubble_sort.go
package main

import "fmt"

func main() {
	arr := [10]int{9, 0, 8, 1, 7, 2, 6, 3, 5, 4}

	for i := 0; i < len(arr); i++ {
		for j := 0; j < len(arr)-i-1; j++ {
			if arr[j] > arr[j+1] {
				arr[j], arr[j+1] = arr[j+1], arr[j]
			}
		}
	}

	fmt.Println(arr)
}
